allennlp/models/coreference_resolution/coref.py

```diff
@@ -505,7 +728,7 @@ def _compute_span_pair_embeddings(
             (batch_size, num_spans_to_keep, max_antecedents, embedding_size).
         antecedent_offsets : `torch.IntTensor`, required.
             The offsets between each top span and its antecedent spans in terms
-            of spans we are considering. Has shape (1, max_antecedents).
+            of spans we are considering. Has shape (batch_size, num_spans_to_keep, max_antecedents).
```

Comment:
I don't understand this change - was this comment wrong? Why do we need batch size here again? Is it because the span pair embeddings are no longer symmetric or something?

Reply:
Yes, for the non-coarse-to-fine case, we always select the antecedents with the same offsets, i.e. `[max_antecedents, 0)`. But the coarse-to-fine case allows more flexibility so that each span can select arbitrary antecedents with a scoring function.